File: Back/NodeJS/modules/bets/src/infrastructure/repositories/sql/SqlFriendshipRepository.ts
import { DomainEventAccessor } from "../../../../../shared/infrastructure/events/DomainEventAccessor";
import { Friendship } from "../../../domain/friendships/Friendship";
import { IFriendshipRepository } from "../../../domain/friendships/IFriendshipRepository";
import { BetContext } from "../../DataAccess/BetContext";
import { FriendshipEntity } from "../../DataAccess/entities/FriendshipEntity";

export class SqlFriendshipRepository implements IFriendshipRepository {
    
    constructor(private readonly betContext: BetContext, 
                    private readonly domainEventAccessor: DomainEventAccessor
        ){}
        
    async SaveAsync(friendship: Friendship): Promise<void> {
        const entity = FriendshipEntity.Create(friendship);
        await this.betContext.FriendshipEntity.save(entity);
        this.domainEventAccessor.AddEvents(friendship.DomainEvents);
    }

}

File: Back/.Net/Modules/Bets/BetFriends.Bets.Infrastructure/Repositories/Sql/SqlFriendshipRepository.cs
using BetFriends.Bets.Domain.Friends;
using BetFriends.Bets.Domain.Friendships;
using BetFriends.Bets.Infrastructure.Repositories.Sql.DataAccess;
using BetFriends.Shared.Infrastructure.Event;

namespace BetFriends.Bets.Infrastructure.Repositories.Sql;

public class SqlFriendshipRepository(BetContext betContext,
                                        DomainEventsAccessor domainEventsAccessor) : IFriendshipRepository
{
    private readonly BetContext betContext = betContext;
    private readonly DomainEventsAccessor domainEventsAccessor = domainEventsAccessor;

    public async Task SaveAsync(Friendship friendship)
    {
        var entity = new FriendshipEntity(friendship);
        await betContext.AddAsync(entity);
        domainEventsAccessor.Add(friendship.Events);
    }
}
